pkg/cli/project/project.go

```diff
@@ -320,13 +320,22 @@ func (o *ProjectOptions) GetContextFromName(contextName string) (*clientcmdapi.C
 }
 
 func ConfirmProjectAccess(currentProject string, projectClient projectv1client.ProjectV1Interface, kClient corev1client.CoreV1Interface) error {
+	projects, err := GetProjects(projectClient, kClient)
```

Comment:
`GetProjects` calls Projects.List which you might not have a permission to do while you could still have one for Projects.Get. Though, not sure if it's a real scenario since usually when you are given Get perms, you get List as well.

Reply:
I don't like we list all projects and if that fails we get the actual projects, this seems backwards. `ConfirmProjectAccess` should try a get on a project and alternatively on a namespace, that's it. I wouldn't change any code in it. It seems overly complicated. It's the caller business to handle the error accordingly to what I said in the comment yesterday.